modules/home/my_home/default.nix

```diff
@@ -24,114 +23,111 @@ with lib;
       };
     };
   };
-    config = mkIf (cfg.username != "") {
-      home-manager = {
-        #TODO: sort out whether we want global or user packages
-        # useUserPackages = true;
-        useGlobalPkgs = true;
-        #TODO: Consider pulling the username from the first key in cfg.users.user
-        #  basically all of my systems are only defined for a single user
-        users.${cfg.username} = {
-          xdg.systemDirs.data = ["$HOME/.nix-profile/share"];
 
-          programs = {
-            starship = {
-              enable = true;
-            };
-            # Let Home Manager install and manage itself.
-            home-manager.enable = true;
-            bash = {
-              enable = true;
-              enableCompletion = true;
-            };
-            bat.enable = true;
-            command-not-found.enable = true;
-            direnv = {
-              enable = true;
-              enableBashIntegration = true;
-              nix-direnv.enable = true;
-            };
-            fzf = {
-              enable = true;
-              enableBashIntegration = true;
-            };
-            gpg.enable = true;
-            htop.enable = true;
-            jq.enable = true;
-            less.enable = true;
-            git = {
-              userEmail = cfg.git.email;
-              userName = cfg.git.username;
-              enable = true;
-              aliases = {
-                c = "commit";
-                co = "checkout";
-                s = "status";
-              };
-              extraConfig = {
-                init.defaultBranch = "main";
-                pull = {
-                  rebase = true;
-                };
-                protocol = {
-                  http.allow = "never";
-                  git.allow = "never";
-                };
-                credential.helper = "store";
-                push = {
-                  autoSetupRemote = true;
-                  default = "current";
-                };
-              };
-            };
-          };
-          editorconfig = {
-            enable = true;
-            settings = {
-              "*" = {
-                indent_size = 2;
-                indent_style = "space";
-              };
-            };
-          };
-          home = {
-            username = cfg.username;
-            homeDirectory = "/home/${cfg.username}";
+  config = {
+    xdg.systemDirs = mkIf pkgs.stdenv.isLinux {
```

Comment:
Some Home Manager options are platform-specific. One way to check is using `pkgs.stdenv.isLinux` and `pkgs.stdenv.isDarwin`. Alternatively, Snowfall Lib passes a lot of helpful args to your module that you _could_ use instead. Though if you're sharing things then you should prefer the NixPkgs-native solutions.